packages/@aws-cdk/aws-ecs-patterns/lib/base/application-load-balanced-service-base.ts

```diff
@@ -273,17 +280,21 @@ export abstract class ApplicationLoadBalancedServiceBase extends cdk.Construct {
 
     this.loadBalancer = props.loadBalancer !== undefined ? props.loadBalancer : new ApplicationLoadBalancer(this, 'LB', lbProps);
 
-    const targetProps = {
-      port: 80
-    };
-
     if (props.certificate !== undefined && props.protocol !== undefined && props.protocol !== ApplicationProtocol.HTTPS) {
       throw new Error('The HTTPS protocol must be used when a certificate is given');
     }
     const protocol = props.protocol !== undefined ? props.protocol : (props.certificate ? ApplicationProtocol.HTTPS : ApplicationProtocol.HTTP);
 
+    const listenerPort = props.listenerPort !== undefined ? props.listenerPort : protocol === ApplicationProtocol.HTTPS ? 443 : 80;
```

Comment:
I like what you've done here, it might be worth opening an issue to track this.

The issue is this is a breaking change. If a user has a production service running with HTTPS but using the previous construct, they would currently be running on 80, not 443. 

`const listenerPort = props.listenerPort !== undefined ? props.listenerPort : 80;`

Reply:
OK created https://github.com/aws/aws-cdk/issues/4861

Reply:
You should update the check to be `const listenerPort = props.listenerPort !== undefined ? props.listenerPort : 80;` This module is stable and we cannot accept a breaking change.

Reply:
OK. I got it reverted.

Reply:
I probably need this otherwise I will break a [test](https://github.com/aws/aws-cdk/pull/4825#issuecomment-549938000).

That is, when no `listenerPort` specified, pass it as `undefined` to avoid the breaking.

```js
 const listenerPort = props.listenerPort !== undefined ? props.listenerPort : undefined;

    const targetProps = {
      port: listenerPort,
      protocol
    };

    this.listener = this.loadBalancer.addListener('PublicListener', {
      protocol,
      port: listenerPort,
      open: true
    });
    this.targetGroup = this.listener.addTargets('ECS', targetProps);
```